### PR TITLE
Add support for MOI.TimeLimitSec

### PR DESCRIPTION
**Pull request by matbesancon**

### CODE REVIEW

**Comment by matbesancon** [comment]
tackles #279

**Comment by matbesancon** [comment]
Time limit is set correctly as checked in JuMP:
```
------------------------------------------------------------------
	      SCS v3.2.3 - Splitting Conic Solver
	(c) Brendan O'Donoghue, Stanford University, 2012
------------------------------------------------------------------
problem:  variables n: 3, constraints m: 8
cones: 	 l: linear vars: 8
settings: eps_abs: 1.0e-04, eps_rel: 1.0e-04, eps_infeas: 1.0e-07
	 alpha: 1.50, scale: 1.00e-01, adaptive_scale: 1
	 max_iters: 100000, normalize: 1, rho_x: 1.00e-06
	 acceleration_lookback: 10, acceleration_interval: 10
	 time_limit_secs: 1.00e+01
lin-sys:  sparse-direct-amd-qdldl
	 nnz(A): 12, nnz(P): 0
------------------------------------------------------------------
 iter | pri res | dua res |   gap   |   obj   |  scale  | time (s)
------------------------------------------------------------------
     0| 1.00e+01  1.00e+00  2.99e+01 -1.49e+01  1.00e-01  1.46e-04 
    75| 6.68e-04  1.63e-05  7.30e-05 -3.00e+01  1.00e-01  2.04e-04 
------------------------------------------------------------------
status:  solved
timings: total: 2.06e-04s = setup: 1.03e-04s + solve: 1.04e-04s
	lin-sys: 1.65e-05s, cones: 6.95e-06s, accel: 4.02e-06s
------------------------------------------------------------------
objective = -29.999936
------------------------------------------------------------------
```

**Comment by mlubin** [comment]
Is the correct status returned when SCS hits the time limit?

**Comment by matbesancon** [comment]
Good point, I updated the status accordingly and added both iteration and time limit, they are both under "inaccurate" for SCS and required checking the raw status. Just tested both with JuMP and both worked

**Comment by matbesancon** [comment]
```
julia> optimize!(m)
------------------------------------------------------------------
	      SCS v3.2.3 - Splitting Conic Solver
	(c) Brendan O'Donoghue, Stanford University, 2012
------------------------------------------------------------------
problem:  variables n: 2000, constraints m: 4002
cones: 	 l: linear vars: 4002
settings: eps_abs: 1.0e-04, eps_rel: 1.0e-04, eps_infeas: 1.0e-07
	 alpha: 1.50, scale: 1.00e-01, adaptive_scale: 1
	 max_iters: 5, normalize: 1, rho_x: 1.00e-06
	 acceleration_lookback: 10, acceleration_interval: 10
lin-sys:  sparse-direct-amd-qdldl
	 nnz(A): 6000, nnz(P): 0
------------------------------------------------------------------
 iter | pri res | dua res |   gap   |   obj   |  scale  | time (s)
------------------------------------------------------------------
     0| 2.04e+01  2.01e+00  1.83e+04 -9.28e+03  1.00e-01  3.77e-03 
     5| 5.77e+00  5.49e-01  2.92e+03 -6.66e+03  1.00e-01  4.44e-03 
------------------------------------------------------------------
status:  solved (inaccurate - reached max_iters)
timings: total: 4.49e-03s = setup: 3.38e-03s + solve: 1.11e-03s
	lin-sys: 3.64e-04s, cones: 5.58e-05s, accel: 1.81e-07s
------------------------------------------------------------------
objective = -6660.726521 (inaccurate)
------------------------------------------------------------------

julia> termination_status(m)
ITERATION_LIMIT::TerminationStatusCode = 11
```

and 
```
julia> optimize!(m)
------------------------------------------------------------------
	      SCS v3.2.3 - Splitting Conic Solver
	(c) Brendan O'Donoghue, Stanford University, 2012
------------------------------------------------------------------
problem:  variables n: 2000, constraints m: 4002
cones: 	 l: linear vars: 4002
settings: eps_abs: 1.0e-04, eps_rel: 1.0e-04, eps_infeas: 1.0e-07
	 alpha: 1.50, scale: 1.00e-01, adaptive_scale: 1
	 max_iters: 100000, normalize: 1, rho_x: 1.00e-06
	 acceleration_lookback: 10, acceleration_interval: 10
	 time_limit_secs: 1.00e-02
lin-sys:  sparse-direct-amd-qdldl
	 nnz(A): 6000, nnz(P): 0
------------------------------------------------------------------
 iter | pri res | dua res |   gap   |   obj   |  scale  | time (s)
------------------------------------------------------------------
     0| 2.70e+01  2.01e+00  9.55e+03 -4.94e+03  1.00e-01  1.59e-02 
    50| 4.33e-01  2.95e-02  5.29e-01 -7.89e+03  1.00e-01  2.88e-02 
------------------------------------------------------------------
status:  solved (inaccurate - reached time_limit_secs)
timings: total: 2.89e-02s = setup: 1.46e-02s + solve: 1.43e-02s
	lin-sys: 8.72e-03s, cones: 1.24e-03s, accel: 3.91e-04s
------------------------------------------------------------------
objective = -7885.401148 (inaccurate)
------------------------------------------------------------------

julia> termination_status(m)
TIME_LIMIT::TerminationStatusCode = 12

```

**Comment by matbesancon** [comment]
@mlubin if you can trigger the workflows that would be great, I don't have access on this repo

File: src/MOI_wrapper/MOI_wrapper.jl [hunk 1]
```diff
@@ -172,6 +172,24 @@ end
...
 
+###
+### MOI.TimeLimitSec
+###
+
+MOI.supports(::Optimizer, ::MOI.TimeLimitSec) = true
+
+function MOI.set(optimizer::Optimizer, ::MOI.TimeLimitSec, time_limit)
+    optimizer.options[:time_limit_secs] = time_limit
+    return nothing
```
Comment (odow) [inline src/MOI_wrapper/MOI_wrapper.jl:183]:
```suggestion
    return
```

**Comment by odow** [comment]
Tweaked the implementation to handle `nothing` being passed.

**Comment by matbesancon** [comment]
@odow setting nothing could be a good test to add to `MOIT`

**Comment by odow** [comment]
Ah. There's already a test. But I see an issue with it

**Pull request merged**